src/metabase/query_processor/middleware/optimize_temporal_filters.clj

```diff
@@ -111,6 +111,36 @@
      (temporal-value :guard optimizable-temporal-value?)]
     (field-and-temporal-value-have-compatible-units? field temporal-value)))
 
+(defn- not-default-bucket-clause
+  [clause]
+  (and (vector? clause)
+       (not= :default (get-in clause [2 :temporal-unit]))))
+
+;; TODO: I believe we do not generate __filter clauses that have default temporal bucket on column arg which should be
+;;       optimized__. Unfortunately I'm not certain about that. If I was, the following `can-optimize-filter? :>=` and
+;;       `can-optimize-filter? :>=` definitions would be redundant after update of `optimizable-expr?`, ie. changing
+;;       the logic to something along "if `expr` has default temporal unit we should not optimize".
+
+(defmethod can-optimize-filter? :>=
+  [filter-clause]
+  (lib.util.match/match-one
+   filter-clause
+   [_tag
+    ;; Don't optimize >= with column that has default temporal bucket
+    (field :guard (every-pred not-default-bucket-clause optimizable-expr?))
+    (temporal-value :guard optimizable-temporal-value?)]
+   (field-and-temporal-value-have-compatible-units? field temporal-value)))
+
+(defmethod can-optimize-filter? :<
+  [filter-clause]
+  (lib.util.match/match-one
+   filter-clause
+   [_tag
+    ;; Don't optimize < with column that has default temporal bucket
+    (field :guard (every-pred not-default-bucket-clause optimizable-expr?))
+    (temporal-value :guard optimizable-temporal-value?)]
+   (field-and-temporal-value-have-compatible-units? field temporal-value)))
+
```

Comment:
This is required to avoid optimization of already optimized expressions generated by desugaring `realtive-time-interval`.